Userland/Libraries/LibJIT/GDBElf.cpp

```diff
@@ -26,7 +26,7 @@ Optional<FixedArray<u8>> build_gdb_image(ReadonlyBytes code, StringView file_sym
     auto const text = sections.build_nobits([&](Elf64_Shdr& text) {
         text.sh_name = section_names.insert(".text"sv);
         text.sh_flags = SHF_EXECINSTR | SHF_ALLOC;
-        text.sh_addr = bit_cast<u64>(code.offset(0));
+        text.sh_addr = bit_cast<uintptr_t>(code.offset(0));
```

Comment:
I am never sure if `uintptr_t` or `FlatPtr` is preferred, especially when it comes to ELF stuff
(in doubt `decltype<text.shaddr>`)

Reply:
`text.sh_addr` is `Elf64_Addr` aka `u64`. That doesn't compile when the pointer is 32-bit; we're supposed to cast it to a pointer-sized (32-bit) integer first, and then to `u64`. That's what I've tried to do here.

Better still would be to keep 32-bit ELF support and use `ElfW(Addr)`, but I doubt anyone else would prefer that.

Reply:
Shrug from me. I would generally prefer FlatPtr though. Keeping the macros around feels a bit icky though. I thought we were working to remove them.